aws/elbv2_test.go

```diff
@@ -25,20 +25,36 @@ func createTestELBv2(t *testing.T, session *session.Session, name string) elbv2.
 		assert.Fail(t, "Needs at least 2 subnets to create ELBv2")
 	}
 
-	subnet1 := *subnetOutput.Subnets[0]
-	subnet2 := *subnetOutput.Subnets[1]
+	var firstSubnet ec2.Subnet
+	var secondSubnet ec2.Subnet
+
+	// ensure we select subnets in different availability zones
+	for _, subnet1 := range subnetOutput.Subnets {
+		for _, subnet2 := range subnetOutput.Subnets {
+			if subnet1.AvailabilityZone != subnet2.AvailabilityZone && subnet1.SubnetId != subnet2.SubnetId {
```

Comment:
If `ec2.Subnet` is a struct, then all of these are empty strings initially, aren't they? And if it's a pointer, won't this cause a `nil` pointer dereference?

Reply:
Yeah, good catch. I should be checking that the pointers aren't nil and then failing the tests if they are